### PR TITLE
Bump referenced Velero tag.

### PR DESCRIPTION
**Pull request by blackpiglet**

### CODE REVIEW

**Review by ywk253100 — APPROVED** [review]

**Review by shubham-pampattiwar — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -3,83 +3,85 @@ module github.com/vmware-tanzu/velero-plugin-for-gcp
...
+	github.com/sirupsen/logrus v1.9.0
 	github.com/spf13/pflag v1.0.5
-	github.com/stretchr/testify v1.7.0
-	github.com/vmware-tanzu/velero v1.7.1
-	golang.org/x/oauth2 v0.0.0-20211104180415-d3ed0bb246c8
-	google.golang.org/api v0.63.0
-	k8s.io/api v0.22.2
-	k8s.io/apimachinery v0.22.2
+	github.com/stretchr/testify v1.8.2
+	github.com/vmware-tanzu/velero v0.0.0-20230727074327-a6d79fc272a2
```
Comment (kaovilai) [inline go.mod:12]: why not a specific tagged release? like `v1.11.1`?

Comment (kaovilai) [reply go.mod:12]: would be easier to read and match with tagged commits in the future.

Comment (sseago) [reply go.mod:12]: @kaovilai this is for the main branch, not release-1.11 -- ongoing dev work for a release that will come out with 1.12. Not sure about this plugin, but certainly the CSI pluign needs post-1.11 stuff. On the release branch, we can peg this to a specific release tag, but for ongoing dev work, that's not always possible. This is just the lastest main commit as of when this PR was submitted.

Comment (kaovilai) [reply go.mod:12]: ahh yes.

**Review by kaovilai — APPROVED** [review]

**Comment by kaovilai** [comment]
I imagine this is something CI (github actions) could do on a regular basis to check for breakage?

**Pull request merged**